wgsl/extract-grammar.py

```diff
@@ -3,28 +3,50 @@
 from datetime import date
 from string import Template
 
+import argparse
```

Comment:
This file is restructured so its jobs can be run in separate phases.
It also adds one more phase, which is to run "unit tests" from wgsl_unit_tests.py
That lets us check the treesitter parser against many tiny examples *not* in the WGSL spec.  This is enormously helpful.

I use python argument parsing to help specify a "flow" to run, which is a set of letters representing the 4 interesting phases.  See how the invocation is changed, in the Makefile.